settings.py

```diff
@@ -504,9 +504,8 @@ def JINJA_CONFIG():
     from django.core.cache.backends.memcached import MemcachedCache
     from django.core.cache import caches
     cache = caches['memcache']
-    config = {'extensions': ['jinja2.ext.i18n', 'tower.template.i18n',
-                             'jinja2.ext.with_', 'jinja2.ext.loopcontrols',
-                             'jinja2.ext.autoescape',
+    config = {'extensions': ['jinja2.ext.i18n', 'jinja2.ext.with_',
```

Comment:
gettext doesn't get added to the context by the `jinja2.ext.i18n` extension or the `puente.ext.PuenteI18nExtension'. It gets added by calling `.install_gettext_translations()` on the environment.

Tower did that here:

https://github.com/clouserw/tower/blob/master/tower/__init__.py#L90

It had to use its own gettext/ngettext.

Jingo does it here:

https://github.com/jbalogh/jingo/blob/v0.7.1/jingo/__init__.py#L69

and specifically looks for `jinja2.ext.i18n` because awesome.

So it looks like if we take out Tower and you're using Jingo, then you do need to have `jinja2.ext.i18n`. I'd make it the first extension with `puente.ext.PuenteI18nExtension` right after (like you had). Hopefully that stomps on the right things.

See if that works. I think you said it did before. If it does, then I'll update the Puente docs. Sorry about that.